File: app-android/src/androidTest/java/com/ustadmobile/port/android/view/XapiPackageContentActivityEspressoTest.java
package com.ustadmobile.port.android.view;


import android.content.Intent;
import android.os.SystemClock;
import android.support.test.InstrumentationRegistry;
import android.support.test.espresso.intent.rule.IntentsTestRule;
import android.support.v4.content.ContextCompat;

import com.ustadmobile.core.db.UmAppDatabase;
import com.ustadmobile.core.view.XapiPackageContentView;
import com.ustadmobile.port.sharedse.util.UmFileUtilSe;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;

import java.io.File;
import java.io.IOException;

import static android.support.test.espresso.web.assertion.WebViewAssertions.webContent;
import static android.support.test.espresso.web.matcher.DomMatchers.hasElementWithId;
import static android.support.test.espresso.web.sugar.Web.onWebView;


public class XapiPackageContentActivityEspressoTest {

    @Rule
    public IntentsTestRule<XapiPackageContentActivity> mActivityRule =
            new IntentsTestRule<>(XapiPackageContentActivity.class, false, false);


    private UmAppDatabase db;

    private UmAppDatabase repo;

    private UmFileUtilSe.TempZipContainer tempXapiPackageContainer;

    @Before
    public void setup() throws IOException {
        db = UmAppDatabase.getInstance(InstrumentationRegistry.getTargetContext());
        repo = db.getRepository("http://localhost/dummy/", "");
        db.clearAllTables();

        File storageDir = ContextCompat.getExternalFilesDirs(
                InstrumentationRegistry.getTargetContext(), null)[0];
        File containerTmpDir = new File(storageDir, "XapiPackageCOntentActivityEspressoTest."
            + System.currentTimeMillis());
        containerTmpDir.mkdirs();

        tempXapiPackageContainer = UmFileUtilSe.makeTempContainerFromClassResource(db, repo,
                "/com/ustadmobile/port/android/view/XapiPackage-JsTetris_TCAPI.zip",
                containerTmpDir);
    }

    @After
    public void tearDown() throws IOException{
        if(tempXapiPackageContainer != null)
            UmFileUtilSe.deleteRecursively(tempXapiPackageContainer.getContainerFileDir());
    }

    public void launchActivity() {
        Intent launchIntent = new Intent();
        launchIntent.putExtra(XapiPackageContentView.ARG_CONTAINER_UID,
                String.valueOf(tempXapiPackageContainer.getContainer().getContainerUid()));
        mActivityRule.launchActivity(launchIntent);
    }

    @Test
    public void givenValidXapiZip_whenCreated_thenShouldShowContents() {
        launchActivity();
        SystemClock.sleep(1000);
        onWebView().check(webContent(hasElementWithId("tetris")));
    }


}

File: app-android/src/androidTest/java/com/ustadmobile/port/android/view/XapiPackageContentActivityEspressoTest.kt
package com.ustadmobile.port.android.view


import android.content.Intent
import android.os.SystemClock
import androidx.core.content.ContextCompat
import androidx.test.espresso.intent.rule.IntentsTestRule
import androidx.test.espresso.web.assertion.WebViewAssertions.webContent
import androidx.test.espresso.web.matcher.DomMatchers.hasElementWithId
import androidx.test.espresso.web.sugar.Web.onWebView
import androidx.test.platform.app.InstrumentationRegistry
import com.ustadmobile.core.db.UmAppDatabase
import com.ustadmobile.core.view.XapiPackageContentView
import com.ustadmobile.port.sharedse.util.UmFileUtilSe
import org.junit.After
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.w3c.dom.Document
import java.io.File
import java.io.IOException


class XapiPackageContentActivityEspressoTest {

    @get:Rule
    var mActivityRule = IntentsTestRule(XapiPackageContentActivity::class.java, false, false)

    private var db: UmAppDatabase? = null

    private var repo: UmAppDatabase? = null

    private var tempXapiPackageContainer: UmFileUtilSe.TempZipContainer? = null

    private var context = InstrumentationRegistry.getInstrumentation().context

    @Before
    @Throws(IOException::class)
    fun setup() {
        db = UmAppDatabase.getInstance(context)
        repo = db //db!!.getUmRepository("http://localhost/dummy/", "")
        db!!.clearAllTables()

        val storageDir = ContextCompat.getExternalFilesDirs(
                context, null)[0]
        val containerTmpDir = File(storageDir, "XapiPackageCOntentActivityEspressoTest." + System.currentTimeMillis())
        containerTmpDir.mkdirs()

        tempXapiPackageContainer = UmFileUtilSe.makeTempContainerFromClassResource(db!!, repo!!,
                "/com/ustadmobile/port/android/view/XapiPackage-JsTetris_TCAPI.zip",
                containerTmpDir)
    }

    @After
    @Throws(IOException::class)
    fun tearDown() {
        if (tempXapiPackageContainer != null)
            UmFileUtilSe.deleteRecursively(tempXapiPackageContainer!!.containerFileDir)
    }

    fun launchActivity() {
        val launchIntent = Intent()
        launchIntent.putExtra(XapiPackageContentView.ARG_CONTAINER_UID,
                tempXapiPackageContainer!!.container.containerUid.toString())
        mActivityRule.launchActivity(launchIntent)
    }

    @Test
    fun givenValidXapiZip_whenCreated_thenShouldShowContents() {
        launchActivity()
        SystemClock.sleep(1000)
        onWebView().check<Document>(webContent(hasElementWithId("tetris")))
    }


}
